packages/spektate/src/Validation.ts

```diff
@@ -0,0 +1,349 @@
+            if (storagePartitionKey !== "") {
+              resolve();
+            } else {
+              resolve({
+                message: "Partition key needs to be non-empty"
+              });
+            }
+          } else {
+            resolve({
+              message: err.message,
+              stack: err.stack
+            });
+          }
+        }
+      );
+    });
+  } catch (e) {
+    return {
+      message: e.message
+    };
+  }
+};
+
+/**
+ * Validates that pipeline is accessible using pipeline personal access token
+ * @param org Azure DevOps organization name
+ * @param project Azure DevOps project name
+ * @param pat Azure DevOps pipeline access token
+ */
+export const verifyPipeline = async (
+  org: string,
+  project: string,
+  pat: string
+): Promise<IValidationError | undefined> => {
+  try {
+    if (org === "" || project === "") {
+      throw new Error("Organization and/or project name are not defined");
+    }
+    const pipeline = new AzureDevOpsPipeline(org, project, pat);
+    await pipeline.getListOfBuilds();
+    return undefined;
+  } catch (e) {
+    return {
+      message:
+        "Pipeline org, project or personal access token are invalid. " +
+        e.toString()
+    };
+  }
+};
+
+/**
+ * Validates that manifest repo is accessible for fetching cluster sync statuses
+ * @param repoName repository name
+ * @param pat personal access token for manifest repo
+ * @param org Azure DevOps organization name
+ * @param project Azure DevOps project name
+ * @param githubUsername GitHub organization or username for the manifest
+ * repository, eg. for github.com/microsoft/bedrock.git, this would be "microsoft"
+ */
+export const verifyManifestRepo = async (
+  repoName: string,
+  pat: string,
+  org: string,
+  project: string,
+  githubUsername?: string
+): Promise<IValidationError | undefined> => {
+  return new Promise<IValidationError | undefined>(async (resolve, reject) => {
+    let repo: IAzureDevOpsRepo | IGitHub | undefined;
+    if (githubUsername && githubUsername !== "") {
+      repo = {
+        reponame: repoName,
+        username: githubUsername
+      };
+      await getGitHubClusterSync(repo, pat)
+        .then(tags => {
+          resolve();
+        })
+        .catch(e => {
+          resolve({
+            message:
+              "Failed to verify manifest repo for cluster sync status. " +
+              e.toString()
+          });
+        });
+    } else if (repoName !== "") {
+      repo = {
+        org,
+        project,
+        repo: repoName
+      };
+      await getADOClusterSync(repo, pat)
+        .then(tags => {
+          resolve();
+        })
+        .catch(e => {
+          resolve({
+            message:
+              "Failed to verify manifest repo for cluster sync status. " +
+              e.toString()
+          });
+        });
+    } else {
+      resolve({
+        message: "Manifest repository could not be recognized. "
+      });
+    }
+  });
+};
```

Comment:
i would write this function like this to avoid the `then` and `catch`. I other functions too, `then` and `catch` makes code reading more difficult (https://dev.to/gafi/7-reasons-to-always-use-async-await-over-plain-promises-tutorial-4ej9) Maybe you want to seek @evanlouie opinion on this. 
```
export const verifyManifestRepo = async (
  repoName: string,
  pat: string,
  org: string,
  project: string,
  githubUsername?: string
): Promise<IValidationError | undefined> => {
  try {
    if (githubUsername) {
      await getGitHubClusterSync({
        reponame: repoName,
        username: githubUsername
      }, pat);
      return undefined;
    } else if (repoName) {
      await getADOClusterSync({
        org,
        project,
        repo: repoName
      }, pat);
      return undefined;
    } else {
      return {
        message: "Manifest repository could not be recognized."
      };
    }
  } catch(err) {
    return {
      message:
        "Failed to verify manifest repo for cluster sync status. " +
        e.toString()
    };
  }  
};
```